code/espurna/light.ino

```diff
@@ -469,6 +469,39 @@ void _lightProviderUpdate() {
 // PERSISTANCE
 // -----------------------------------------------------------------------------
 
+union rtcmem_light_t {
+    struct {
+        uint8_t channels[5];
+        uint8_t brightness;
+        uint16_t mired;
+    } packed;
+    uint64_t value;
+};
+
+void _rtcmemLightSave() {
+    rtcmem_light_t light;
+    for (unsigned int i=0; i < lightChannels(); i++) {
```

Comment:
We don't support any device with more than 5 channels (are there any?) but maybe we should check here the `lightChannels()` return value.

Reply:
Yeah, that is true.
Unlike pwm channels init this cannot really depend on LIGHT_CHANNELS to init array size. `rtcmem_light_t::channels` size is still hardcoded, but this can be `std::min(lightChannels(), sizeof(rtcmem_light_t::channels))` instead of simply using it.